ush/setup.py

```diff
@@ -1000,637 +981,195 @@ def get_location(xcs, fmt):
-    # necessary module files and runs the tasks.
+    # Check that the set of tasks to run in the workflow is internally
+    # consistent.
     #
     # -----------------------------------------------------------------------
     #
-    global LOAD_MODULES_RUN_TASK_FP
-    LOAD_MODULES_RUN_TASK_FP = os.path.join(USHdir, "load_modules_run_task.sh")
-
-    global RUN_TASK_MAKE_GRID, RUN_TASK_MAKE_OROG, RUN_TASK_MAKE_SFC_CLIMO
-    global RUN_TASK_VX_GRIDSTAT, RUN_TASK_VX_POINTSTAT, RUN_TASK_VX_ENSGRID, RUN_TASK_VX_ENSPOINT
-
-    # Fix file location
-    if RUN_TASK_MAKE_GRID:
-        FIXdir = EXPTDIR
-    else:
-        FIXdir = os.path.join(HOMEdir, "fix")
-
-    FIXam = os.path.join(FIXdir, "fix_am")
-    FIXclim = os.path.join(FIXdir, "fix_clim")
-    FIXlam = os.path.join(FIXdir, "fix_lam")
+    workflow_switches = expt_config["workflow_switches"]
 
     # Ensemble verification can only be run in ensemble mode
-    if (not DO_ENSEMBLE) and (RUN_TASK_VX_ENSGRID or RUN_TASK_VX_ENSPOINT):
+    do_ensemble = global_sect["DO_ENSEMBLE"]
+    run_task_vx_ensgrid = workflow_switches["RUN_TASK_VX_ENSGRID"]
+    run_task_vx_enspoint = workflow_switches["RUN_TASK_VX_ENSPOINT"]
+    if (not do_ensemble) and (run_task_vx_ensgrid or run_task_vx_enspoint):
         raise Exception(
-            f"""
+            f'''
             Ensemble verification can not be run unless running in ensemble mode:
-               DO_ENSEMBLE = '{DO_ENSEMBLE}'
-               RUN_TASK_VX_ENSGRID = '{RUN_TASK_VX_ENSGRID}'
-               RUN_TASK_VX_ENSPOINT = '{RUN_TASK_VX_ENSPOINT}'"""
+               DO_ENSEMBLE = \"{do_ensemble}\"
+               RUN_TASK_VX_ENSGRID = \"{run_task_vx_ensgrid}\"
+               RUN_TASK_VX_ENSPOINT = \"{run_task_vx_enspoint}\"'''
         )
 
     #
-    # -----------------------------------------------------------------------
-    #
-    # Define the various work subdirectories under the main work directory.
-    # Each of these corresponds to a different step/substep/task in the pre-
-    # processing, as follows:
-    #
-    # GRID_DIR:
-    # Directory in which the grid files will be placed (if RUN_TASK_MAKE_GRID
-    # is set to True) or searched for (if RUN_TASK_MAKE_GRID is set to
-    # False).
-    #
-    # OROG_DIR:
-    # Directory in which the orography files will be placed (if RUN_TASK_MAKE_OROG
-    # is set to True) or searched for (if RUN_TASK_MAKE_OROG is set to
-    # False).
-    #
-    # SFC_CLIMO_DIR:
-    # Directory in which the surface climatology files will be placed (if
-    # RUN_TASK_MAKE_SFC_CLIMO is set to True) or searched for (if
-    # RUN_TASK_MAKE_SFC_CLIMO is set to False).
-    #
-    # ----------------------------------------------------------------------
-    #
-    global GRID_DIR, OROG_DIR, SFC_CLIMO_DIR
-
-    if DOMAIN_PREGEN_BASEDIR is None:
-        RUN_TASK_MAKE_GRID = True
-        RUN_TASK_MAKE_OROG = True
-        RUN_TASK_MAKE_SFC_CLIMO = True
-
```

Comment:
The logic for this has moved up to line 444, and has also been changed to raise an exception to tell the user that there's no way to get the files, instead of assuming this is what should be done.